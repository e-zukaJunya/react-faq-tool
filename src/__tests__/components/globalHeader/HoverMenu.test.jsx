import HoverMenu from "components/globalHeader/HoverMenu";
import { mount } from "enzyme";
import React from "react";
import { MemoryRouter } from "react-router-dom";

// jsdomにcreateRangeを実装
document.createRange = () => ({
  setStart: () => {},
  setEnd: () => {},
  commonAncestorContainer: {
    nodeName: "BODY",
    ownerDocument: document
  }
});

describe("AlertDialog", () => {
  const tab = {
    category: "KB",
    pages: [
      { id: "kbAdd", name: "追加" },
      { id: "kbEdit", name: "編集" },
      { id: "kbApply", name: "適用" }
    ]
  };

  beforeEach(() => {});

  test("mount", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/"]}>
        <HoverMenu tab={tab} />
      </MemoryRouter>
    );
    // expect(wrapper.debug()).toMatchSnapshot();
    const dom = wrapper.getDOMNode();
    expect(dom).toMatchSnapshot();

    const tabButton = wrapper.find("div[role='button']");
    expect(tabButton.text()).toBe(tab.category);
  });

  test("open", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/"]}>
        <HoverMenu tab={tab} />
      </MemoryRouter>
    );
    wrapper.simulate("mouseenter");

    const dom = wrapper.getDOMNode();
    expect(dom).toMatchSnapshot();

    const menuButtons = wrapper.find("li");
    expect(menuButtons.length).toBe(tab.pages.length);

    const receiveTexts = menuButtons.map(item => item.text());
    const expectTexts = tab.pages.map(item => item.name);
    expect(receiveTexts).toStrictEqual(expectTexts);
  });

  test("close", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/"]}>
        <HoverMenu tab={tab} />
      </MemoryRouter>
    );
    wrapper.simulate("mouseenter");
    wrapper.simulate("mouseleave");
    const dom = wrapper.getDOMNode();
    // todo menuが完全に消えていない（opacityは0）
    // アニメーションが完了していないせい？
    expect(dom).toMatchSnapshot();
  });

  test("click tab", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/"]}>
        <HoverMenu tab={tab} />
      </MemoryRouter>
    );
    const tabButton = wrapper.find("div[role='button']");
    tabButton.simulate("click");
    const dom = wrapper.getDOMNode();
    expect(dom).toMatchSnapshot();
  });

  test("click menu", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/"]}>
        <HoverMenu tab={tab} />
      </MemoryRouter>
    );
    wrapper.simulate("mouseenter");
    const menuButtons = wrapper.find("li");
    const targetBtn = menuButtons.at(0);
    targetBtn.simulate("click");
    const props = wrapper.childAt(0).props();
    expect(props.history.location.pathname).toContain(tab.pages[0].id);
  });
});
