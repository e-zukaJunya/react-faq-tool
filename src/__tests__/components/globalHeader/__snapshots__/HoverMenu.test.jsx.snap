// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDialog click tab 1`] = `
<div>
  <div
    aria-controls="menu-list-grow"
    aria-disabled="false"
    aria-haspopup="true"
    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    role="button"
    tabindex="0"
  >
    KB
    <span
      class="MuiTouchRipple-root"
    />
  </div>
  <div
    role="tooltip"
    style="position: absolute;"
    x-placement="bottom"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      style="opacity: 1; transform: scale(1, 1); transform-origin: center bottom; transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <ul
        class="MuiList-root MuiList-padding"
        id="menu-list-grow"
        role="menu"
        tabindex="-1"
      >
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="0"
        >
          追加
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="-1"
        >
          編集
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="-1"
        >
          適用
          <span
            class="MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`AlertDialog close 1`] = `
<div>
  <div
    aria-disabled="false"
    aria-haspopup="true"
    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    role="button"
    tabindex="0"
  >
    KB
    <span
      class="MuiTouchRipple-root"
    />
  </div>
  <div
    role="tooltip"
    style="position: absolute;"
    x-placement="bottom"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      style="opacity: 0; transform: scale(0.75, 0.5625); transform-origin: center bottom; transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <ul
        class="MuiList-root MuiList-padding"
        id="menu-list-grow"
        role="menu"
        tabindex="-1"
      >
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="0"
        >
          追加
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="-1"
        >
          編集
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="-1"
        >
          適用
          <span
            class="MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`AlertDialog mount 1`] = `
<div>
  <div
    aria-disabled="false"
    aria-haspopup="true"
    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    role="button"
    tabindex="0"
  >
    KB
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;

exports[`AlertDialog open 1`] = `
<div>
  <div
    aria-controls="menu-list-grow"
    aria-disabled="false"
    aria-haspopup="true"
    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    role="button"
    tabindex="0"
  >
    KB
    <span
      class="MuiTouchRipple-root"
    />
  </div>
  <div
    role="tooltip"
    style="position: absolute;"
    x-placement="bottom"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      style="opacity: 1; transform: scale(1, 1); transform-origin: center bottom; transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <ul
        class="MuiList-root MuiList-padding"
        id="menu-list-grow"
        role="menu"
        tabindex="-1"
      >
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="0"
        >
          追加
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="-1"
        >
          編集
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="-1"
        >
          適用
          <span
            class="MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
